packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -0,0 +1,182 @@
+
+interface DynamoDBItem {
+  RequestID: string;
+  PrescriptionID: string;
+  PatientNHSNumber: string;
+  PharmacyODSCode: string;
+  TaskID: string;
+  LineItemID: string;
+  TerminalStatusIndicator: string;
+  LastModified: string;
+  Timestamp: string;
+  RequestMessage: any;
+}
+
+const lambdaHandler = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    console.log("Received event:", event)
+    const requestBody = JSON.parse(event.body || "")
+    console.log("Parsed request body:", requestBody)
+
+    const entries = requestBody.entry
+    console.log("Entries:", entries)
+
+    if (!entries || entries.length === 0) {
+      console.log("Missing required fields")
+      return {
+        statusCode: 400,
+        body: JSON.stringify({error: "Missing required fields"}),
+        headers: {
+          "Content-Type": "application/fhir+json",
+          "Cache-Control": "no-cache"
+        }
+      }
+    }
+
+    const responseBundle: any = {
+      resourceType: "Bundle",
+      id: uuidv4(),
+      meta: {
+        lastUpdated: new Date().toISOString()
+      },
+      type: "transaction-response",
+      entry: []
+    }
+
+    for (const entry of entries) {
+      console.log("Processing entry:", entry)
+      const entry_resource = entry.resource
+      console.log("Entry resource:", entry_resource)
+
+      const dynamoDBItem: DynamoDBItem = {
+        RequestID: "",
+        PrescriptionID: "",
+        PatientNHSNumber: "",
+        PharmacyODSCode: "",
+        TaskID: "",
+        LineItemID: "",
+        TerminalStatusIndicator: "",
+        LastModified: "",
+        Timestamp: "",
+        RequestMessage: {}
+      }
+
+      dynamoDBItem.RequestID = uuidv4()
+      dynamoDBItem.PrescriptionID =
+        entry_resource.basedOn?.[0]?.identifier?.value
+      dynamoDBItem.PatientNHSNumber = entry_resource.for?.identifier?.value
+      dynamoDBItem.PharmacyODSCode = entry_resource.owner?.identifier?.value
+      dynamoDBItem.TaskID = entry_resource.id
+      dynamoDBItem.LineItemID = entry_resource.focus?.identifier?.value
+      dynamoDBItem.TerminalStatusIndicator = entry_resource.status
+      dynamoDBItem.LastModified = entry_resource.lastModified
+      dynamoDBItem.Timestamp = new Date().toISOString()
+      dynamoDBItem.RequestMessage = entry_resource
+
+      const invalidFields = []
+      for (const field in dynamoDBItem) {
+        if (!dynamoDBItem[field as keyof DynamoDBItem]) {
+          invalidFields.push(field)
+        }
+      }
+
+      if (invalidFields.length > 0) {
+        const errorMessage = `Missing required fields: ${invalidFields.join(
+          ", "
+        )}`
+        console.log(errorMessage)
+        return {
+          statusCode: 400,
+          body: JSON.stringify({error: errorMessage}),
+          headers: {
+            "Content-Type": "application/json",
+            "Cache-Control": "no-cache"
+          }
+        }
+      }
```

Comment:
The error response still doesn't appear to be a valid fhir response with an operation outcome

Reply:
A proper validation will be provided with the tickets that follows.

Reply:
The structure of the response has no bearing on any validation, it should still be a valid fhir response as per the spec. Leaving it as is would not be conforming to fhir regardless of what validation triggers this to be returned.

Reply:
The change has been implemented.
```
        const errorResponseBody = {
          resourceType: "OperationOutcome",
          issue: [
            {
              code: "value",
              severity: "error",
              details: {
                coding: [
                  {
                    system: "https://fhir.nhs.uk/CodeSystem/Spine-ErrorOrWarningCode",
                    code: "MISSING_VALUE",
                    display: errorMessage
                  }
                ]
              }
            }
          ]
        }
        return {
          statusCode: 400,
          body: JSON.stringify(errorResponseBody),
          headers: {
            "Content-Type": "application/fhir+json",
            "Cache-Control": "no-cache"
          }
        }
      }
```